File: packages/capacitor/ios/Sources/EodinCapacitorPlugin/EodinCapacitorAnalyticsPlugin.swift
import Capacitor

@objc(EodinCapacitorAnalyticsPlugin)
public class EodinCapacitorAnalyticsPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "EodinCapacitorAnalyticsPlugin"
    public let jsName = "EodinAnalytics"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "configure", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "track", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "identify", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "clearIdentity", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "setAttribution", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "flush", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "startSession", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "endSession", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "requestTrackingAuthorization", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "getATTStatus", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "getStatus", returnType: CAPPluginReturnPromise),
        // GDPR (Phase 1.7 — open-issues §4.5)
        CAPPluginMethod(name: "setEnabled", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "isEnabled", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "requestDataDeletion", returnType: CAPPluginReturnPromise),
    ]

    @objc func configure(_ call: CAPPluginCall) {
        guard let apiEndpoint = call.getString("apiEndpoint"),
              let apiKey = call.getString("apiKey"),
              let appId = call.getString("appId") else {
            call.reject("apiEndpoint, apiKey, and appId are required")
            return
        }

        EodinAnalytics.configure(
            apiEndpoint: apiEndpoint,
            apiKey: apiKey,
            appId: appId,
            debug: call.getBool("debug") ?? false,
            offlineMode: call.getBool("offlineMode") ?? true
        )
        call.resolve()
    }

    @objc func track(_ call: CAPPluginCall) {
        guard let eventName = call.getString("eventName") else {
            call.reject("eventName is required")
            return
        }
        let properties = call.getObject("properties") as? [String: Any]
        EodinAnalytics.track(eventName, properties: properties)
        call.resolve()
    }

    @objc func identify(_ call: CAPPluginCall) {
        guard let userId = call.getString("userId") else {
            call.reject("userId is required")
            return
        }
        EodinAnalytics.identify(userId)
        call.resolve()
    }

    @objc func clearIdentity(_ call: CAPPluginCall) {
        EodinAnalytics.clearIdentity()
        call.resolve()
    }

    @objc func setAttribution(_ call: CAPPluginCall) {
        let attribution = Attribution(
            source: call.getString("source"),
            campaignId: call.getString("campaignId"),
            adsetId: call.getString("adsetId"),
            adId: call.getString("adId"),
            clickId: call.getString("clickId"),
            clickIdType: call.getString("clickIdType"),
            utmSource: call.getString("utmSource"),
            utmMedium: call.getString("utmMedium"),
            utmCampaign: call.getString("utmCampaign"),
            utmContent: call.getString("utmContent"),
            utmTerm: call.getString("utmTerm")
        )
        EodinAnalytics.setAttribution(attribution)
        call.resolve()
    }

    @objc func flush(_ call: CAPPluginCall) {
        EodinAnalytics.flush()
        call.resolve()
    }

    @objc func startSession(_ call: CAPPluginCall) {
        EodinAnalytics.startSession()
        call.resolve()
    }

    @objc func endSession(_ call: CAPPluginCall) {
        EodinAnalytics.endSession()
        call.resolve()
    }

    @objc func requestTrackingAuthorization(_ call: CAPPluginCall) {
        EodinAnalytics.requestTrackingAuthorization { status in
            call.resolve(["status": status.rawValue])
        }
    }

    @objc func getATTStatus(_ call: CAPPluginCall) {
        call.resolve(["status": EodinAnalytics.attStatus.rawValue])
    }

    @objc func getStatus(_ call: CAPPluginCall) {
        call.resolve([
            "isConfigured": EodinAnalytics.isConfigured,
            "deviceId": EodinAnalytics.deviceId as Any,
            "userId": EodinAnalytics.userId as Any,
            "sessionId": EodinAnalytics.sessionId as Any,
            "isOnline": EodinAnalytics.isOnline,
            "queueSize": EodinAnalytics.queueSize,
            "attStatus": EodinAnalytics.attStatus.rawValue
        ])
    }

    // MARK: - GDPR (Phase 1.7 — open-issues §4.5)

    @objc func setEnabled(_ call: CAPPluginCall) {
        guard let enabled = call.getBool("enabled") else {
            call.reject("enabled (bool) is required")
            return
        }
        EodinAnalytics.setEnabled(enabled)
        call.resolve()
    }

    @objc func isEnabled(_ call: CAPPluginCall) {
        call.resolve(["enabled": EodinAnalytics.isEnabled])
    }

    @objc func requestDataDeletion(_ call: CAPPluginCall) {
        EodinAnalytics.requestDataDeletion { success in
            call.resolve(["success": success])
        }
    }
}

File: packages/capacitor/android/src/main/java/app/eodin/capacitor/EodinCapacitorAnalyticsPlugin.kt
package app.eodin.capacitor

import com.getcapacitor.JSObject
import com.getcapacitor.Plugin
import com.getcapacitor.PluginCall
import com.getcapacitor.PluginMethod
import com.getcapacitor.annotation.CapacitorPlugin
import app.eodin.analytics.EodinAnalytics
import app.eodin.analytics.models.Attribution

@CapacitorPlugin(name = "EodinAnalytics")
class EodinCapacitorAnalyticsPlugin : Plugin() {

    @PluginMethod
    fun configure(call: PluginCall) {
        val apiEndpoint = call.getString("apiEndpoint")
        val apiKey = call.getString("apiKey")
        val appId = call.getString("appId")

        if (apiEndpoint == null || apiKey == null || appId == null) {
            call.reject("apiEndpoint, apiKey, and appId are required")
            return
        }

        EodinAnalytics.configure(
            context = context,
            apiEndpoint = apiEndpoint,
            apiKey = apiKey,
            appId = appId,
            debug = call.getBoolean("debug", false) ?: false,
            offlineMode = call.getBoolean("offlineMode", true) ?: true
        )
        call.resolve()
    }

    @PluginMethod
    fun track(call: PluginCall) {
        val eventName = call.getString("eventName")
        if (eventName == null) {
            call.reject("eventName is required")
            return
        }

        val properties = call.getObject("properties")?.let { obj ->
            val map = mutableMapOf<String, Any>()
            obj.keys().forEach { key -> obj.opt(key)?.let { map[key] = it } }
            map
        }

        EodinAnalytics.track(eventName, properties)
        call.resolve()
    }

    @PluginMethod
    fun identify(call: PluginCall) {
        val userId = call.getString("userId")
        if (userId == null) {
            call.reject("userId is required")
            return
        }
        EodinAnalytics.identify(userId)
        call.resolve()
    }

    @PluginMethod
    fun clearIdentity(call: PluginCall) {
        EodinAnalytics.clearIdentity()
        call.resolve()
    }

    @PluginMethod
    fun setAttribution(call: PluginCall) {
        val attribution = Attribution(
            source = call.getString("source"),
            campaignId = call.getString("campaignId"),
            adsetId = call.getString("adsetId"),
            adId = call.getString("adId"),
            clickId = call.getString("clickId"),
            clickIdType = call.getString("clickIdType"),
            utmSource = call.getString("utmSource"),
            utmMedium = call.getString("utmMedium"),
            utmCampaign = call.getString("utmCampaign"),
            utmContent = call.getString("utmContent"),
            utmTerm = call.getString("utmTerm")
        )
        EodinAnalytics.setAttribution(attribution)
        call.resolve()
    }

    @PluginMethod
    fun flush(call: PluginCall) {
        EodinAnalytics.flush()
        call.resolve()
    }

    @PluginMethod
    fun startSession(call: PluginCall) {
        EodinAnalytics.startSession()
        call.resolve()
    }

    @PluginMethod
    fun endSession(call: PluginCall) {
        EodinAnalytics.endSession()
        call.resolve()
    }

    @PluginMethod
    fun requestTrackingAuthorization(call: PluginCall) {
        val data = JSObject()
        data.put("status", "authorized")
        call.resolve(data)
    }

    @PluginMethod
    fun getATTStatus(call: PluginCall) {
        val data = JSObject()
        data.put("status", "authorized")
        call.resolve(data)
    }

    @PluginMethod
    fun getStatus(call: PluginCall) {
        val data = JSObject().apply {
            put("isConfigured", EodinAnalytics.isConfigured)
            put("deviceId", EodinAnalytics.getDeviceId())
            put("userId", EodinAnalytics.getUserId())
            put("sessionId", EodinAnalytics.getSessionId())
            put("isOnline", EodinAnalytics.isOnline())
            put("queueSize", EodinAnalytics.getQueueSize())
            put("attStatus", "authorized")
        }
        call.resolve(data)
    }

    // GDPR (Phase 1.7 — open-issues §4.5)

    @PluginMethod
    fun setEnabled(call: PluginCall) {
        val enabled = call.getBoolean("enabled")
        if (enabled == null) {
            call.reject("enabled (bool) is required")
            return
        }
        EodinAnalytics.setEnabled(enabled)
        call.resolve()
    }

    @PluginMethod
    fun isEnabled(call: PluginCall) {
        call.resolve(JSObject().apply { put("enabled", EodinAnalytics.isEnabled) })
    }

    @PluginMethod
    fun requestDataDeletion(call: PluginCall) {
        EodinAnalytics.requestDataDeletion { success ->
            call.resolve(JSObject().apply { put("success", success) })
        }
    }
}
